ui/jobs/combo_tracker.ts

```diff
@@ -41,15 +52,15 @@ export default class ComboTracker {
       this.AbortCombo(id);
   }
 
-  StateTransition(id, nextState) {
+  StateTransition(id: string | null, nextState?: { id?: string; next?: StartMap } | null): void {
```

Comment:
```suggestion
  StateTransition(id: string | null, nextState: StartMap[string] | null): void {
```

I think `nextState` is always defined, and maybe it's possible to not repeat the type of `StartMap` as well.  That also removes a bunch of `?.` and `??` from down below.